include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -127,7 +87,13 @@ PointwiseOutputGeneratorBase<dim, Number>::setup_base(
 #ifdef DEAL_II_WITH_HDF5
   pointwise_output_data = pointwise_output_data_in;
 
-  if(pointwise_output_data.write_output && pointwise_output_data.evaluation_points.size() > 0)
+  AssertThrow(pointwise_output_data_in.time_control_data.counter_start == 0,
```

Comment:
what is `counter_start`? This looks also like a solution for a special problem inserted into a rather generic data structure.

Reply:
It's the counter's initial value.  I think it is generic to be able to specify the initial value of the counter, even though one might not need it often.

Reply:
See arguments given here: https://github.com/exadg/exadg/pull/262#discussion_r1000353552

So it is a redundancy. A user does not know whether he was to set `counter_start` correctly in case of a restart, even though this is done automatically inside the code. For me, clarity from a user perspective must be a primary goal. Also, you do not take `counter_start` into account when resetting the counter in `needs_evaluation()`. This is at least an inconsistency/bug, even if we agreed to introduce `counter_start`.